include/camp/resource/cuda.hpp

```diff
@@ -129,20 +148,18 @@ namespace resources
         // related: https://stackoverflow.com/questions/64523302/cuda-missing-return-statement-at-end-of-non-void-function-in-constexpr-if-fun
         return MemoryAccess::Unknown;
       }
+
     public:
-      Cuda(int group = -1, int dev = 0)
-          : stream(get_a_stream(group)), device(dev)
+      Cuda(int group = -1, int dev = get_current_device())
+          : stream(get_a_stream(group, dev)), device(dev)
       {
       }
 
       /// Create a resource from a custom stream
       /// The device specified must match the stream, if none is specified the
       /// currently selected device is used.
-      static Cuda CudaFromStream(cudaStream_t s, int dev = -1)
+      static Cuda CudaFromStream(cudaStream_t s, int dev = get_current_device())
```

Comment:
All the rest of this seems good to me. If we're going to support multiple devices this way though, this should be pulling the device out of the context associated with the stream so we don't end up with a mismatched pair.

Reply:
Good idea, I haven't used the driver API before, but this looks pretty straight forward to accomplish.

Reply:
That can be done with cuda, but I don't see how I can find the device of a hip stream.

Reply:
Take a look at https://rocm-developer-tools.github.io/HIP/classihipStream__t.html

It has a `getDevice` member function that returns an internal struct of theirs with a device id in it.

Reply:
It looks like rocm 5 has a hipGetStreamDeviceId function, I'll give that a try https://docs.amd.com/bundle/HIP-API-Guide-v5.3/page/a00196.html#gad16a23519ee4d7d87c7be53004b76608.